go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/cluster-capi-operator
 
-go 1.21
+go 1.22.0
+
+toolchain go1.23.0
```

Comment:
Do we actually want this to be 1.23?

Reply:
IIRC running `make vendor` (I think just `go mod vendor`) will automatically add the toolchain directive now